pkg/operator/quorumguardcontroller/quorumguardcontroller.go

```diff
@@ -0,0 +1,248 @@
+		podLister:            kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Pods().Lister(),
+		nodeLister:           kubeInformers.InformersFor("").Core().V1().Nodes().Lister(),
+		infrastructureLister: infrastructureLister,
+		replicaCount:         0,
+	}
+	return factory.New().ResyncEvery(time.Minute).WithInformers(
+		kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Pods().Informer(),
+		kubeInformers.InformersFor("").Core().V1().Nodes().Informer(),
+		operatorClient.Informer(),
+	).WithSync(c.sync).ResyncEvery(time.Minute).ToController("QuorumGuardController", eventRecorder.WithComponentSuffix("quorum-guard-controller"))
+}
+
+func (c *QuorumGuardController) isInHATopologyMode() (bool, error) {
+	// update once
+	if c.clusterTopology != "" {
+		return c.clusterTopology == configv1.HighlyAvailableTopologyMode, nil
+	}
+
+	var err error
+	c.clusterTopology, err = ceohelpers.GetControlPlaneTopology(c.infrastructureLister)
+	if err != nil {
+		return false, err
+	}
+
+	klog.Infof("HA mode is: %s", c.clusterTopology)
+
+	return c.clusterTopology == configv1.HighlyAvailableTopologyMode, nil
+}
+
+func (c *QuorumGuardController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	err := c.ensureEtcdGuard(ctx, syncCtx.Recorder())
+	if err != nil {
+		_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    "QuorumGuardControllerDegraded",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "Error",
+			Message: err.Error(),
+		}))
+		if updateErr != nil {
+			syncCtx.Recorder().Warning("QuorumGuardControllerUpdatingStatus", updateErr.Error())
+		}
+		return err
+	}
+
+	_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient,
+		v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:   "QuorumGuardControllerDegraded",
+			Status: operatorv1.ConditionFalse,
+			Reason: "AsExpected",
+		}))
+	return updateErr
+}
+
+func (c *QuorumGuardController) ensureEtcdGuard(ctx context.Context, recorder events.Recorder) error {
+	haTopologyMode, err := c.isInHATopologyMode()
+	if err != nil {
+		klog.Infof("Failed to validate ha mode %v ", err)
+		return err
+	}
+	if !haTopologyMode {
+		return nil
+	}
+
+	replicaCount, err := c.getMastersReplicaCount(ctx)
+	if err != nil {
+		return err
+	}
+
+	if err := c.ensureEtcdGuardDeployment(ctx, replicaCount, recorder); err != nil {
+		return err
+	}
+
+	if err := c.ensureEtcdGuardPdbDeployment(ctx, recorder); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ensureEtcdGuardDeployment if etcd quorum guard deployment doesn't exist - add it, else skip
+func (c *QuorumGuardController) ensureEtcdGuardDeployment(ctx context.Context, replicaCount int32, recorder events.Recorder) error {
+	_, err := c.kubeClient.AppsV1().Deployments(operatorclient.TargetNamespace).Get(ctx,
+		EtcdGuardDeploymentName, metav1.GetOptions{})
+	switch {
+	case apierrors.IsNotFound(err):
```

Comment:
we need to not only ensure it exists but that its correct. If user makes a change to the PDB etc we want to stop it 

```
if !reflect.DeepEqual(observed, expected) {
  //stomp user trying to do naughty things and replace proper values for deployment and pdb
}
```